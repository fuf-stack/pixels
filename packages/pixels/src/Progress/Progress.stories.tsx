import type { ProgressProps } from '@heroui/progress';
import type { Meta, StoryObj } from '@storybook/react';

import Progress, { progressVariants } from './Progress';

const meta: Meta<typeof Progress> = {
  title: 'pixels/Progress',
  component: Progress,
  decorators: [
    (Story) => (
      <div style={{ width: '300px' }}>
        <Story />
      </div>
    ),
  ],
} as Meta<typeof Progress>;

export default meta;
type Story = StoryObj<typeof Progress>;

export const Default: Story = {};

export const WithLabel: Story = {
  args: {
    label: 'label',
  },
};

export const NoValueLabel: Story = {
  args: {
    showValueLabel: false,
  },
};

export const CompletedSuccess: Story = {
  render: (args) => (
    <>
      <Progress {...args} percent={42} successOnComplete />
      <Progress {...args} percent={100} successOnComplete />
      <Progress {...args} percent={150} successOnComplete />
    </>
  ),
  args: {},
};

export const AllStyleVariants: Story = {
  render: (args) => (
    <>
      {Object.keys(progressVariants.variants.color).map((color) => (
        <div key={color} style={{ marginTop: '10px' }}>
          <Progress
            {...args}
            label={color}
            percent={42}
            color={color as ProgressProps['color']}
          />
        </div>
      ))}
    </>
  ),
  argTypes: {
    // do not show styleVariant in controls table
    color: {
      table: {
        disable: true,
      },
    },
  },
};

/** SOME PERCENTAGES */
const percent = [0, 10, 20, 30, 40, 50, 60, 70, 80, 90, 100, 150] as number[];
export const SomePercentages: Story = {
  render: (args) => (
    <>
      {percent.map((p) => (
        <div key={p} style={{ marginTop: '10px' }}>
          <Progress {...args} percent={p} />
        </div>
      ))}
    </>
  ),
  argTypes: {
    // do not show percent in controls table
    percent: {
      table: {
        disable: true,
      },
    },
  },
};
