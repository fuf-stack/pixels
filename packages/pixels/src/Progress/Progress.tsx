import type { TVClassName, TVProps } from '@fuf-stack/pixel-utils';
import type { ReactNode } from 'react';

import { Progress as NextProgress } from '@nextui-org/progress';

import { tv, variantsToClassNames } from '@fuf-stack/pixel-utils';

// progress styling variants
export const progressVariants = tv({
  slots: {
    base: '', // wrapper around the whole component
    indicator: '', // indicator of the progress (finished part)
    label: '', // label at the top left of the progress bar
    labelWrapper: '', // wrapper around progress label and label span
    track: '', // track of the progress (not finished part)
    value: '', // span around the progress value
  },
  variants: {
    // See: https://github.com/nextui-org/nextui/blob/canary/packages/core/theme/src/components/progress.ts
    color: {
      danger: {
        indicator: 'bg-danger',
      },
      default: {
        indicator: 'bg-default-400',
      },
      info: {
        indicator: 'bg-info',
      },
      primary: {
        indicator: 'bg-primary',
      },
      secondary: {
        indicator: 'bg-secondary',
      },
      success: {
        indicator: 'bg-success',
      },
      warning: {
        indicator: 'bg-warning',
      },
    },
  },
});

export type VariantProps = TVProps<typeof progressVariants>;
type ClassName = TVClassName<typeof progressVariants>;

export interface ProgressProps extends VariantProps {
  /** CSS class name */
  className?: ClassName;
  /** color of the progress bar */
  color?: VariantProps['color'];
  /** function to format the progress value */
  format?: (percent: number) => ReactNode;
  /** label of the progress bar */
  label?: ReactNode;
  /** percentage/progress of the progress bar */
  percent: number;
  /** whether to show the value label */
  showValueLabel?: boolean;
  /** size of the progress bar */
  size?: 'sm' | 'md' | 'lg' | undefined;
  /** whether to show success color on complete */
  successOnComplete?: boolean;
  /** HTML data-testid attribute used in e2e tests */
  testId?: string;
}

const defaultFormat = (percent: number) => `${percent}%`;

/**
 * Progress component based on [NextUI Progress](https://nextui.org/docs/components/progress)
 */
const Progress = ({
  className = undefined,
  color = 'primary',
  format = defaultFormat,
  label = undefined,
  percent,
  showValueLabel = true,
  size = 'md',
  successOnComplete = false,
  testId = undefined,
}: ProgressProps) => {
  const processedColor =
    successOnComplete && percent >= 100 ? 'success' : color;

  const variants = progressVariants({ color: processedColor });
  const classNames = variantsToClassNames(variants, className, 'base');

  const value = format(percent);

  return (
    <NextProgress
      aria-label="Loading..."
      classNames={classNames}
      data-testid={testId}
      formatOptions={undefined}
      label={label}
      showValueLabel={showValueLabel}
      size={size}
      value={percent || 0}
      valueLabel={percent && value}
    />
  );
};

export default Progress;
