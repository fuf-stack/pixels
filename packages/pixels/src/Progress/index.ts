import Progress, { progressVariants } from './Progress';

export type { ProgressProps } from './Progress';

export { Progress, progressVariants };

export default Progress;
