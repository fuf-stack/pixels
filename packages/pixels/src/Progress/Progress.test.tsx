import { describe } from 'vitest';

import storySnapshots from '@repo/storybook-config/story-snapshots';

import * as stories from './Progress.stories';

describe('Story Snapshots', () => {
  storySnapshots(stories);
});
