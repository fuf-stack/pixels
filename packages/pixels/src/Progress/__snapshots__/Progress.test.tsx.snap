// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Story Snapshots > AllColors 1`] = `
<div>
  <div
    style="width: 300px;"
  >
    <div
      style="margin-top: 10px;"
    >
      <div
        aria-label="progress"
        aria-labelledby="react-aria-react-useId-mock"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="42"
        aria-valuetext="42%"
        class="flex flex-col gap-2 w-full"
        id="react-aria-react-useId-mock"
        role="progressbar"
      >
        <div
          class="flex justify-between"
        >
          <span
            class="text-medium"
            id="react-aria-react-useId-mock"
          >
            danger
          </span>
        </div>
        <div
          class="z-0 relative bg-default-300/50 overflow-hidden rtl:rotate-180 h-3 rounded-full"
        >
          <div
            class="h-full rounded-full transition-transform !duration-500 bg-danger"
            style="transform: translateX(-100%);"
          />
        </div>
      </div>
    </div>
    <div
      style="margin-top: 10px;"
    >
      <div
        aria-label="progress"
        aria-labelledby="react-aria-react-useId-mock"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="42"
        aria-valuetext="42%"
        class="flex flex-col gap-2 w-full"
        id="react-aria-react-useId-mock"
        role="progressbar"
      >
        <div
          class="flex justify-between"
        >
          <span
            class="text-medium"
            id="react-aria-react-useId-mock"
          >
            default
          </span>
        </div>
        <div
          class="z-0 relative bg-default-300/50 overflow-hidden rtl:rotate-180 h-3 rounded-full"
        >
          <div
            class="h-full rounded-full transition-transform !duration-500 bg-default-400"
            style="transform: translateX(-100%);"
          />
        </div>
      </div>
    </div>
    <div
      style="margin-top: 10px;"
    >
      <div
        aria-label="progress"
        aria-labelledby="react-aria-react-useId-mock"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="42"
        aria-valuetext="42%"
        class="flex flex-col gap-2 w-full"
        id="react-aria-react-useId-mock"
        role="progressbar"
      >
        <div
          class="flex justify-between"
        >
          <span
            class="text-medium"
            id="react-aria-react-useId-mock"
          >
            info
          </span>
        </div>
        <div
          class="z-0 relative bg-default-300/50 overflow-hidden rtl:rotate-180 h-3 rounded-full"
        >
          <div
            class="h-full rounded-full transition-transform !duration-500 bg-info"
            style="transform: translateX(-100%);"
          />
        </div>
      </div>
    </div>
    <div
      style="margin-top: 10px;"
    >
      <div
        aria-label="progress"
        aria-labelledby="react-aria-react-useId-mock"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="42"
        aria-valuetext="42%"
        class="flex flex-col gap-2 w-full"
        id="react-aria-react-useId-mock"
        role="progressbar"
      >
        <div
          class="flex justify-between"
        >
          <span
            class="text-medium"
            id="react-aria-react-useId-mock"
          >
            primary
          </span>
        </div>
        <div
          class="z-0 relative bg-default-300/50 overflow-hidden rtl:rotate-180 h-3 rounded-full"
        >
          <div
            class="h-full rounded-full transition-transform !duration-500 bg-primary"
            style="transform: translateX(-100%);"
          />
        </div>
      </div>
    </div>
    <div
      style="margin-top: 10px;"
    >
      <div
        aria-label="progress"
        aria-labelledby="react-aria-react-useId-mock"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="42"
        aria-valuetext="42%"
        class="flex flex-col gap-2 w-full"
        id="react-aria-react-useId-mock"
        role="progressbar"
      >
        <div
          class="flex justify-between"
        >
          <span
            class="text-medium"
            id="react-aria-react-useId-mock"
          >
            secondary
          </span>
        </div>
        <div
          class="z-0 relative bg-default-300/50 overflow-hidden rtl:rotate-180 h-3 rounded-full"
        >
          <div
            class="h-full rounded-full transition-transform !duration-500 bg-secondary"
            style="transform: translateX(-100%);"
          />
        </div>
      </div>
    </div>
    <div
      style="margin-top: 10px;"
    >
      <div
        aria-label="progress"
        aria-labelledby="react-aria-react-useId-mock"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="42"
        aria-valuetext="42%"
        class="flex flex-col gap-2 w-full"
        id="react-aria-react-useId-mock"
        role="progressbar"
      >
        <div
          class="flex justify-between"
        >
          <span
            class="text-medium"
            id="react-aria-react-useId-mock"
          >
            success
          </span>
        </div>
        <div
          class="z-0 relative bg-default-300/50 overflow-hidden rtl:rotate-180 h-3 rounded-full"
        >
          <div
            class="h-full rounded-full transition-transform !duration-500 bg-success"
            style="transform: translateX(-100%);"
          />
        </div>
      </div>
    </div>
    <div
      style="margin-top: 10px;"
    >
      <div
        aria-label="progress"
        aria-labelledby="react-aria-react-useId-mock"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="42"
        aria-valuetext="42%"
        class="flex flex-col gap-2 w-full"
        id="react-aria-react-useId-mock"
        role="progressbar"
      >
        <div
          class="flex justify-between"
        >
          <span
            class="text-medium"
            id="react-aria-react-useId-mock"
          >
            warning
          </span>
        </div>
        <div
          class="z-0 relative bg-default-300/50 overflow-hidden rtl:rotate-180 h-3 rounded-full"
        >
          <div
            class="h-full rounded-full transition-transform !duration-500 bg-warning"
            style="transform: translateX(-100%);"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Story Snapshots > AllSizes 1`] = `
<div>
  <div
    style="width: 300px;"
  >
    <div
      style="margin-top: 10px;"
    >
      <div
        aria-label="progress"
        aria-labelledby="react-aria-react-useId-mock"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="42"
        aria-valuetext="42%"
        class="flex flex-col gap-2 w-full"
        id="react-aria-react-useId-mock"
        role="progressbar"
      >
        <div
          class="flex justify-between"
        >
          <span
            class="text-small"
            id="react-aria-react-useId-mock"
          >
            sm
          </span>
        </div>
        <div
          class="z-0 relative bg-default-300/50 overflow-hidden rtl:rotate-180 h-1 rounded-full"
        >
          <div
            class="h-full rounded-full transition-transform !duration-500 bg-info"
            style="transform: translateX(-100%);"
          />
        </div>
      </div>
    </div>
    <div
      style="margin-top: 10px;"
    >
      <div
        aria-label="progress"
        aria-labelledby="react-aria-react-useId-mock"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="42"
        aria-valuetext="42%"
        class="flex flex-col gap-2 w-full"
        id="react-aria-react-useId-mock"
        role="progressbar"
      >
        <div
          class="flex justify-between"
        >
          <span
            class="text-medium"
            id="react-aria-react-useId-mock"
          >
            md
          </span>
        </div>
        <div
          class="z-0 relative bg-default-300/50 overflow-hidden rtl:rotate-180 h-3 rounded-full"
        >
          <div
            class="h-full rounded-full transition-transform !duration-500 bg-info"
            style="transform: translateX(-100%);"
          />
        </div>
      </div>
    </div>
    <div
      style="margin-top: 10px;"
    >
      <div
        aria-label="progress"
        aria-labelledby="react-aria-react-useId-mock"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="42"
        aria-valuetext="42%"
        class="flex flex-col gap-2 w-full"
        id="react-aria-react-useId-mock"
        role="progressbar"
      >
        <div
          class="flex justify-between"
        >
          <span
            class="text-large"
            id="react-aria-react-useId-mock"
          >
            lg
          </span>
        </div>
        <div
          class="z-0 relative bg-default-300/50 overflow-hidden rtl:rotate-180 h-5 rounded-full"
        >
          <div
            class="h-full rounded-full transition-transform !duration-500 bg-info"
            style="transform: translateX(-100%);"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Story Snapshots > Default 1`] = `
<div>
  <div
    style="width: 300px;"
  >
    <div
      aria-label="progress"
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="42"
      aria-valuetext="42%"
      class="flex flex-col gap-2 w-full"
      id="react-aria-react-useId-mock"
      role="progressbar"
    >
      <div
        class="z-0 relative bg-default-300/50 overflow-hidden rtl:rotate-180 h-3 rounded-full"
      >
        <div
          class="h-full rounded-full transition-transform !duration-500 bg-info"
          style="transform: translateX(-100%);"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Story Snapshots > Finished 1`] = `
<div>
  <div
    style="width: 300px;"
  >
    <div
      aria-label="progress"
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="100"
      aria-valuetext="100%"
      class="flex flex-col gap-2 w-full"
      id="react-aria-react-useId-mock"
      role="progressbar"
    >
      <div
        class="z-0 relative bg-default-300/50 overflow-hidden rtl:rotate-180 h-3 rounded-full"
      >
        <div
          class="h-full rounded-full transition-transform !duration-500 bg-success"
          style="transform: translateX(-100%);"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Story Snapshots > Indeterminate 1`] = `
<div>
  <div
    style="width: 300px;"
  >
    <div
      aria-label="progress"
      aria-valuemax="100"
      aria-valuemin="0"
      class="flex flex-col gap-2 w-full"
      data-indeterminate="true"
      id="react-aria-react-useId-mock"
      role="progressbar"
    >
      <div
        class="z-0 relative bg-default-300/50 overflow-hidden rtl:rotate-180 h-3 rounded-full"
      >
        <div
          class="h-full rounded-full absolute w-full origin-left animate-indeterminate-bar transition-transform !duration-500 bg-info"
          style="transform: translateX(-100%);"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Story Snapshots > ShowValueLabel 1`] = `
<div>
  <div
    style="width: 300px;"
  >
    <div
      aria-label="progress"
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="42"
      aria-valuetext="42%"
      class="flex flex-col gap-2 w-full"
      id="react-aria-react-useId-mock"
      role="progressbar"
    >
      <div
        class="flex justify-between"
      >
        <span
          class="text-medium"
        >
          42%
        </span>
      </div>
      <div
        class="z-0 relative bg-default-300/50 overflow-hidden rtl:rotate-180 h-3 rounded-full"
      >
        <div
          class="h-full rounded-full transition-transform !duration-500 bg-info"
          style="transform: translateX(-100%);"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Story Snapshots > SomePercentages 1`] = `
<div>
  <div
    style="width: 300px;"
  >
    <div
      style="margin-top: 10px;"
    >
      <div
        aria-label="progress"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="0"
        aria-valuetext="0%"
        class="flex flex-col gap-2 w-full"
        id="react-aria-react-useId-mock"
        role="progressbar"
      >
        <div
          class="z-0 relative bg-default-300/50 overflow-hidden rtl:rotate-180 h-3 rounded-full"
        >
          <div
            class="h-full rounded-full transition-transform !duration-500 bg-info"
            style="transform: translateX(-100%);"
          />
        </div>
      </div>
    </div>
    <div
      style="margin-top: 10px;"
    >
      <div
        aria-label="progress"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="10"
        aria-valuetext="10%"
        class="flex flex-col gap-2 w-full"
        id="react-aria-react-useId-mock"
        role="progressbar"
      >
        <div
          class="z-0 relative bg-default-300/50 overflow-hidden rtl:rotate-180 h-3 rounded-full"
        >
          <div
            class="h-full rounded-full transition-transform !duration-500 bg-info"
            style="transform: translateX(-100%);"
          />
        </div>
      </div>
    </div>
    <div
      style="margin-top: 10px;"
    >
      <div
        aria-label="progress"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="20"
        aria-valuetext="20%"
        class="flex flex-col gap-2 w-full"
        id="react-aria-react-useId-mock"
        role="progressbar"
      >
        <div
          class="z-0 relative bg-default-300/50 overflow-hidden rtl:rotate-180 h-3 rounded-full"
        >
          <div
            class="h-full rounded-full transition-transform !duration-500 bg-info"
            style="transform: translateX(-100%);"
          />
        </div>
      </div>
    </div>
    <div
      style="margin-top: 10px;"
    >
      <div
        aria-label="progress"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="30"
        aria-valuetext="30%"
        class="flex flex-col gap-2 w-full"
        id="react-aria-react-useId-mock"
        role="progressbar"
      >
        <div
          class="z-0 relative bg-default-300/50 overflow-hidden rtl:rotate-180 h-3 rounded-full"
        >
          <div
            class="h-full rounded-full transition-transform !duration-500 bg-info"
            style="transform: translateX(-100%);"
          />
        </div>
      </div>
    </div>
    <div
      style="margin-top: 10px;"
    >
      <div
        aria-label="progress"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="40"
        aria-valuetext="40%"
        class="flex flex-col gap-2 w-full"
        id="react-aria-react-useId-mock"
        role="progressbar"
      >
        <div
          class="z-0 relative bg-default-300/50 overflow-hidden rtl:rotate-180 h-3 rounded-full"
        >
          <div
            class="h-full rounded-full transition-transform !duration-500 bg-info"
            style="transform: translateX(-100%);"
          />
        </div>
      </div>
    </div>
    <div
      style="margin-top: 10px;"
    >
      <div
        aria-label="progress"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="50"
        aria-valuetext="50%"
        class="flex flex-col gap-2 w-full"
        id="react-aria-react-useId-mock"
        role="progressbar"
      >
        <div
          class="z-0 relative bg-default-300/50 overflow-hidden rtl:rotate-180 h-3 rounded-full"
        >
          <div
            class="h-full rounded-full transition-transform !duration-500 bg-info"
            style="transform: translateX(-100%);"
          />
        </div>
      </div>
    </div>
    <div
      style="margin-top: 10px;"
    >
      <div
        aria-label="progress"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="60"
        aria-valuetext="60%"
        class="flex flex-col gap-2 w-full"
        id="react-aria-react-useId-mock"
        role="progressbar"
      >
        <div
          class="z-0 relative bg-default-300/50 overflow-hidden rtl:rotate-180 h-3 rounded-full"
        >
          <div
            class="h-full rounded-full transition-transform !duration-500 bg-info"
            style="transform: translateX(-100%);"
          />
        </div>
      </div>
    </div>
    <div
      style="margin-top: 10px;"
    >
      <div
        aria-label="progress"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="70"
        aria-valuetext="70%"
        class="flex flex-col gap-2 w-full"
        id="react-aria-react-useId-mock"
        role="progressbar"
      >
        <div
          class="z-0 relative bg-default-300/50 overflow-hidden rtl:rotate-180 h-3 rounded-full"
        >
          <div
            class="h-full rounded-full transition-transform !duration-500 bg-info"
            style="transform: translateX(-100%);"
          />
        </div>
      </div>
    </div>
    <div
      style="margin-top: 10px;"
    >
      <div
        aria-label="progress"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="80"
        aria-valuetext="80%"
        class="flex flex-col gap-2 w-full"
        id="react-aria-react-useId-mock"
        role="progressbar"
      >
        <div
          class="z-0 relative bg-default-300/50 overflow-hidden rtl:rotate-180 h-3 rounded-full"
        >
          <div
            class="h-full rounded-full transition-transform !duration-500 bg-info"
            style="transform: translateX(-100%);"
          />
        </div>
      </div>
    </div>
    <div
      style="margin-top: 10px;"
    >
      <div
        aria-label="progress"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="90"
        aria-valuetext="90%"
        class="flex flex-col gap-2 w-full"
        id="react-aria-react-useId-mock"
        role="progressbar"
      >
        <div
          class="z-0 relative bg-default-300/50 overflow-hidden rtl:rotate-180 h-3 rounded-full"
        >
          <div
            class="h-full rounded-full transition-transform !duration-500 bg-info"
            style="transform: translateX(-100%);"
          />
        </div>
      </div>
    </div>
    <div
      style="margin-top: 10px;"
    >
      <div
        aria-label="progress"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="100"
        aria-valuetext="100%"
        class="flex flex-col gap-2 w-full"
        id="react-aria-react-useId-mock"
        role="progressbar"
      >
        <div
          class="z-0 relative bg-default-300/50 overflow-hidden rtl:rotate-180 h-3 rounded-full"
        >
          <div
            class="h-full rounded-full transition-transform !duration-500 bg-success"
            style="transform: translateX(-100%);"
          />
        </div>
      </div>
    </div>
    <div
      style="margin-top: 10px;"
    >
      <div
        aria-label="progress"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="100"
        aria-valuetext="150%"
        class="flex flex-col gap-2 w-full"
        id="react-aria-react-useId-mock"
        role="progressbar"
      >
        <div
          class="z-0 relative bg-default-300/50 overflow-hidden rtl:rotate-180 h-3 rounded-full"
        >
          <div
            class="h-full rounded-full transition-transform !duration-500 bg-success"
            style="transform: translateX(-100%);"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Story Snapshots > WithLabel 1`] = `
<div>
  <div
    style="width: 300px;"
  >
    <div
      aria-label="progress"
      aria-labelledby="react-aria-react-useId-mock"
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="42"
      aria-valuetext="42%"
      class="flex flex-col gap-2 w-full"
      id="react-aria-react-useId-mock"
      role="progressbar"
    >
      <div
        class="flex justify-between"
      >
        <span
          class="text-medium"
          id="react-aria-react-useId-mock"
        >
          Some Label
        </span>
      </div>
      <div
        class="z-0 relative bg-default-300/50 overflow-hidden rtl:rotate-180 h-3 rounded-full"
      >
        <div
          class="h-full rounded-full transition-transform !duration-500 bg-info"
          style="transform: translateX(-100%);"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Story Snapshots > WithLabelAndValueLabel 1`] = `
<div>
  <div
    style="width: 300px;"
  >
    <div
      aria-label="progress"
      aria-labelledby="react-aria-react-useId-mock"
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="42"
      aria-valuetext="42%"
      class="flex flex-col gap-2 w-full"
      id="react-aria-react-useId-mock"
      role="progressbar"
    >
      <div
        class="flex justify-between"
      >
        <span
          class="text-medium"
          id="react-aria-react-useId-mock"
        >
          Some Label
        </span>
        <span
          class="text-medium"
        >
          42%
        </span>
      </div>
      <div
        class="z-0 relative bg-default-300/50 overflow-hidden rtl:rotate-180 h-3 rounded-full"
      >
        <div
          class="h-full rounded-full transition-transform !duration-500 bg-info"
          style="transform: translateX(-100%);"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Story Snapshots > WithoutPercentProp 1`] = `
<div>
  <div
    style="width: 300px;"
  >
    <div
      aria-label="progress"
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="0"
      aria-valuetext="0%"
      class="flex flex-col gap-2 w-full"
      id="react-aria-react-useId-mock"
      role="progressbar"
    >
      <div
        class="z-0 relative bg-default-300/50 overflow-hidden rtl:rotate-180 h-3 rounded-full"
      >
        <div
          class="h-full rounded-full transition-transform !duration-500 bg-info"
          style="transform: translateX(-100%);"
        />
      </div>
    </div>
  </div>
</div>
`;
